components/automate-cli/pkg/verifyserver/services/softwareversionservice/softwareversionservice.go

```diff
@@ -0,0 +1,220 @@
+package softwareversionservice
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/fiberutils"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/getosutils"
+	"github.com/chef/automate/lib/logger"
+	"github.com/pkg/errors"
+)
+
+type ISoftwareVersionService interface {
+	GetSoftwareVersionDetails(string) (*models.SoftwareVersionDetails, error)
+}
+
+type SoftwareVersionService struct {
+	cmdCheckArray  []string
+	osFilePath     string
+	kernelFilePath string
+	logger         logger.Logger
+	CheckPath      func(cmd string) (string, error)
+}
+
+func NewSoftwareVersionService(logger logger.Logger, CheckPath func(cmd string) (string, error)) ISoftwareVersionService {
+	return &SoftwareVersionService{
+		cmdCheckArray:  cmdCheckArray,
+		osFilePath:     OSFILEPATH,
+		kernelFilePath: KERNELFILEPATH,
+		logger:         logger,
+	}
+}
+
+const (
+	AVAILABILITY              = " availability"
+	ENSURE                    = "Ensure "
+	OSFILEPATH                = "/etc/os-release"
+	KERNELFILEPATH            = "/proc/sys/kernel/osrelease"
+	LINUX_VERSION_CHECK       = "Linux Version Check"
+	KERNAL_VERSION_CHECK      = "Kernal Version Check"
+	UBUNTU                    = "Ubuntu"
+	RED_HAT                   = "Red Hat"
+	OPENSEARCH                = "opensearch"
```

Comment:
line numbers 46 to 50, these are already defined. please use them instead of creating new constants.